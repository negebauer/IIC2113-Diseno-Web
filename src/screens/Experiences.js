import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { devlog } from '../utils/log'
import { fetchExperiences } from '../redux/modules/user'

const mapStateToProps = state => ({
  loading: state.user.loading,
  api_key: state.user.api_key,
  error: state.user.error,
})

const mapDispatchToProps = {
  fetchExperiences,
}

class Experiences extends Component {
  componentWillMount = () => {
    this.props.fetchExperiences(this.state)
  }

  render() {
    devlog('Login', this.props)
    const message =
      (this.props.loading && 'Cargando') ||
      (this.props.error && this.props.error) ||
      'Experiencias'

    
    return (
      <div>
        <div className="container">
          <div className="row">
            <h2 className="header">
              {message}
            </h2>
            <a className="btn-floating btn-large waves-effect waves-light red">
              <i className="material-icons">add</i>
            </a>
            <table className="bordered responsive-table">
              <thead>
                <tr>
                  <th>Name</th>
                  <th>Item Name</th>
                  <th>Item Price</th>
                  <th>Agregar usuario</th>
                </tr>
              </thead>

              <tbody>
                <tr>
                  <td>Alvin</td>
                  <td>Eclair</td>
                  <td>$0.87</td>
                  <td>
                    <a className="waves-effect waves-light btn">
                      <i className="material-icons right">add</i>Agregar
                    </a>
                  </td>
                </tr>
                <tr>
                  <td>Alan</td>
                  <td>Jellybean</td>
                  <td>$3.76</td>
                  <td>
                    <a className="waves-effect waves-light btn">
                      <i className="material-icons right">add</i>Agregar
                    </a>
                  </td>
                </tr>
                <tr>
                  <td>Jonathan</td>
                  <td>Lollipop</td>
                  <td>$7.00</td>
                  <td>
                    <a className="waves-effect waves-light btn">
                      <i className="material-icons right">add</i>Agregar
                    </a>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    )
  }
}

Experiences.propTypes = {
  api_key: PropTypes.string.isRequired,
  error: PropTypes.string.isRequired,
  loading: PropTypes.bool.isRequired,
  fetchExperiences: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Experiences)
