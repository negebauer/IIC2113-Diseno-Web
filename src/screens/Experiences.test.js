import React from 'react'
import createHistory from 'history/createBrowserHistory'
import { Provider } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Experiences from './Experiences'
import configureStore from '../redux/store'

let history
let initialState
let store

beforeEach(() => {
  history = createHistory()
  initialState = {}
  const api = { fetchExperiences: () => {} }
  store = configureStore(initialState, { history, api })
})

it('shallow renders without crashing', () => {
  shallow(<Experiences store={store} />)
})

it('mount renders without crashing', () => {
  mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Experiences />
      </ConnectedRouter>
    </Provider>
  )
})

it('matches snapshot', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Experiences />
      </ConnectedRouter>
    </Provider>
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders children', () => {
  const wrapper = mount(
    <Provider store={store}>
      <ConnectedRouter history={history}>
        <Experiences />
      </ConnectedRouter>
    </Provider>
  )
  expect(wrapper.children().exists()).toBeTruthy()
})
