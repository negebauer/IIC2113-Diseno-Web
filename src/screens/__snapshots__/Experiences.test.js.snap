// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectedRouter
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Connect(Experiences)>
        <Experiences
          api_key=""
          error=""
          experiences={Array []}
          fetchExperiences={[Function]}
          loading={false}
        >
          <div>
            <div
              className="container"
            >
              <div
                className="row"
              >
                <h2
                  className="header"
                >
                  Experiencias
                </h2>
                <span
                  className="btn-floating btn-large waves-effect waves-light red"
                >
                  <i
                    className="material-icons"
                  >
                    <Link
                      replace={false}
                      to="/newexperience"
                    >
                      <a
                        href="/newexperience"
                        onClick={[Function]}
                      >
                        add
                      </a>
                    </Link>
                  </i>
                </span>
                <table
                  className="bordered responsive-table"
                >
                  <thead>
                    <tr>
                      <th>
                        Nombre
                      </th>
                      <th>
                        Descripcion
                      </th>
                      <th>
                        Agregar usuario
                      </th>
                    </tr>
                  </thead>
                  <tbody />
                </table>
              </div>
            </div>
          </div>
        </Experiences>
      </Connect(Experiences)>
    </Router>
  </ConnectedRouter>
</Provider>
`;
